### ISSUE TITLE
`--string --correct` breaks `archlinux-java` when setting PATH

### ISSUE THREAD

**Issue by vgivanovic** [epogrebnyak/justpath#11]
`archlinux-java` is used to switch java versions by changing the directory that `/usr/lib/jvm/default` points to. `--string --correct` follows symbolic links so that `export PATH=$(justpath --string --correct)` might not use the current java version depending on how `archlinux-java` and setting PATH using `justpath --string--correct` are intermixed. `justpath --string --purge-invalid` doesn't follow symbolic links and so allows `archlinux-java` to  continue to switch java versions.

I don't know if following symbolic links is the only difference between `--correct` and `--purge-invalid`.

­---Vladimir
P.S. Thanks for a great utility!

**Comment by epogrebnyak** [comment]
For my own navigation: Path.is_symlink()[¶](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_symlink) this I can use to indicate a directory is symlink. 

Can add --resolve or --follow-simlink flag and set it to false as default. When true the program will resolve all paths.

**Comment by epogrebnyak** [comment]
This is current behavior:

```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --duplicates --includes sbin
35 /usr/sbin (duplicates: 2)
37 /sbin (symlink resolves to /usr/sbin, duplicates: 2)

@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --correct --includes sbin
24 /usr/local/sbin 
26 /usr/sbin 
```

**Comment by epogrebnyak** [comment]
How about this behavoir?

Always indicate a directory is a symlink:
```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --includes sbin 
33 /usr/local/sbin 
35 /usr/sbin 
37 /sbin (symlink resolves to /usr/sbin)
```

When no extra flag do not resolve symlink:
```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --duplicates --includes sbin 
```

with flag do resolve symlinks:
```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --duplicates --includes sbin --follow-symlinks 
35 /usr/sbin (duplicates: 2)
37 /sbin (symlink resolves to /usr/sbin, duplicates: 2)
```

Resolve and purge duplicates:
```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --purge-duplicates --includes sbin --follow-symlinks 
33 /usr/local/sbin 
35 /usr/sbin (duplicates: 2)

@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --correct  --includes sbin --follow-symlinks 
33 /usr/local/sbin 
35 /usr/sbin (duplicates: 2)
```


Do not resole and do not purge if no flag:
```
@epogrebnyak ➜ /workspaces/what-the-path (main) $ justpath --correct  --includes sbin 
33 /usr/local/sbin 
35 /usr/sbin 
37 /sbin (symlink resolves to /usr/sbin)
```

Resolve with flag:
```
justpath --correct --includes sbin --follow-symlinks
33 /usr/local/sbin 
35 /usr/sbin (duplicates: 2)
```

**Comment by epogrebnyak** [comment]
You can new functionality with `pip install justpath==0.0.14`

**Comment by epogrebnyak** [comment]
@vgivanovic do you think I can close the issue now or there is something to be changed?

**Comment by vgivanovic** [comment]
> My apologies. I had wanted to let you know that you could close this issue, but I apparently forgot to let you know.
> 
-Vladimir 

*Vladimir G. Ivanović*
+1 650 450 4101
***@***.***

**Comment by epogrebnyak** [comment]
@vgivanovic - thanks, closing with finalising example from Codespaces

```
$ justpath --duplicates --follow-symlinks
 1 /usr/local/rvm/gems/ruby-3.2.2/bin (duplicates: 2)
 3 /usr/local/rvm/rubies/ruby-3.2.2/bin (duplicates: 3)
 5 /home/codespace/.local/bin (duplicates: 2)
 6 /home/codespace/.dotnet (resolves to /usr/local/dotnet/7.0.306, duplicates: 2)
 7 /home/codespace/nvm/current/bin (duplicates: 2)
 8 /home/codespace/.php/current/bin (duplicates: 2)
 9 /home/codespace/.python/current/bin (duplicates: 2)
10 /home/codespace/java/current/bin (duplicates: 2)
11 /home/codespace/.ruby/current/bin (duplicates: 3)
12 /home/codespace/.local/bin (duplicates: 2)
13 /usr/local/python/current/bin (duplicates: 2)
19 /usr/local/sdkman/candidates/java/current/bin (duplicates: 2)
23 /usr/local/rvm/gems/default/bin (duplicates: 2)
25 /usr/local/rvm/rubies/default/bin (duplicates: 3)
27 /usr/local/php/current/bin (duplicates: 2)
30 /usr/local/share/nvm/versions/node/v20.11.0/bin (duplicates: 2)
32 /usr/local/dotnet/current (resolves to /usr/local/dotnet/7.0.306, duplicates: 2)
35 /usr/sbin (duplicates: 2)
36 /usr/bin (duplicates: 2)
37 /sbin (resolves to /usr/sbin, duplicates: 2)
38 /bin (resolves to /usr/bin, duplicates: 2)
```
